airflow/providers/amazon/aws/hooks/batch_client.py

```diff
@@ -396,7 +396,11 @@ def get_job_description(self, job_id: str) -> dict:
 
             try:
                 response = self.get_conn().describe_jobs(jobs=[job_id])
-                return self.parse_job_description(job_id, response)
+                job_description = self.parse_job_description(job_id, response)
+                # allow us to retry getting the job description in case
+                # we called it before AWS could register the job
+                if job_description:
+                    return job_description
```

Comment:
As mentioned below, instead of changing the spec of the below function to now return None, just catch the AirflowException here and retry on that.

Reply:
Good call, done!